consensus/consortium/v1/consortium.go

```diff
@@ -338,9 +338,19 @@ func (c *Consortium) snapshot(chain consensus.ChainHeaderReader, number uint64,
 			if cpHeader != nil {
 				hash := cpHeader.Hash()
 
-				validators, err := c.getValidatorsFromContract(chain, number)
-				if err != nil {
-					return nil, err
+				var validators []common.Address
+				var err error
+				// Snap Sync mode.
+				if c.skipCheckpointHeaderCheck {
```

Comment:
IMO, I don't think this is needed as getValidatorsFromGenesis must return the same as getValidatorsFromContract at genesis block so we can just use getValidatorsFromGenesis. It's okay to let it like this but I see that in getValidatorsFromLastCheckpoint there is no check like this but use getValidatorsFromGenesis directly. I think we should be consistent here.

Reply:
The first idea is try to limit the impact for snap sync, and keep the remaining logic when we don't use it. But in theory, those infor must be same. Hence, let me remove here for consistency.